scripts/auto-rebase/rebase.sh

```diff
@@ -84,17 +106,43 @@ download_release() {
         repo=$(echo "${line}" | cut -d ' ' -f 2)
         commit=$(echo "${line}" | cut -d ' ' -f 3)
         if [[ "${EMBEDDED_COMPONENTS}" == *"${component}"* ]] || [[ "${LOADED_COMPONENTS}" == *"${component}"* ]] || [[ "${EMBEDDED_COMPONENT_OPERATORS}" == *"${component}"* ]]; then
-            title "## Cloning ${repo} at commit ${commit}..."
-            git init "${repo##*/}"
-            pushd "${repo##*/}" >/dev/null
-            git remote add origin "${repo}"
-            git fetch origin --filter=tree:0 --tags "${commit}"
-            git checkout "${commit}"
-            popd >/dev/null
+            clone_repo "${repo}" "${commit}"
+            echo "${repo} ${commit}" >> "${new_commits_file}"
             echo
         fi
     done < source-commits
 
+    title "# Cloning ${release_image_amd64} image repos"
+
+    # Determine the repos and commits for the repos that build the images
+    cat release_amd64.json \
```

Comment:
Unlike with the embedded dependencies, we may have different sets of changes for different builds of the images. Should we take that into account? How different do we really expect the images to be?

Reply:
Looking at recent discoveries I think one think we might want to take into an account is that amd and arm payloads are promoted to nightlies on different schedules (?) (or rather amd seems to be more frequently rejected).
See:
- https://amd64.ocp.releases.ci.openshift.org/#4.12.0-0.nightly
- https://arm64.ocp.releases.ci.openshift.org/#4.12.0-0.nightly-arm64

Last night's rebase job ran for (we didn't get the PR because 4.12 doesn't include arm in the branch name):
```
./scripts/auto-rebase/rebase.sh to registry.ci.openshift.org/ocp/release:4.12.0-0.nightly-2023-01-10-062211 registry.ci.openshift.org/ocp-arm64/release-arm64:4.12.0-0.nightly-arm64-2023-01-12-221036
```
I don't know where difference comes from but it looks like it's likely that we'd run different commits (i.e. images built of different commits, esp for quickly progressing `4.1{3,4,5,...}` nightly) on both platforms?

Reply:
I compared component version from last [rebase pr](https://github.com/openshift/microshift/pull/1260):
```
amd64: 2023-01-11-225449
arm64: 2023-01-16-225509
```
```
component                                                ARM commit  date                            AMD commit  date                       
route-controller-manager                                 d25295db    "2022-11-29T14:03:09Z"            d25295db    "2022-11-29T14:03:09Z"       
cluster-policy-controller                                f8efde27    "2023-01-12T14:38:56Z"            9d1995d8    "2023-01-09T18:29:36Z"       diff
hyperkube                                                99791b17    "2023-01-09T09:48:17Z"            99791b17    "2023-01-09T09:48:17Z"       
etcd                                                     8785067b    "2023-01-11T21:50:26Z"            94951078    "2022-12-08T11:19:52Z"       diff
cluster-kube-apiserver-operator                          fa1e9be5    "2023-01-16T13:29:28Z"            458ef3c0    "2023-01-10T17:40:01Z"       diff
cluster-kube-controller-manager-operator                 607bf985    "2023-01-16T11:57:57Z"            52572d06    "2023-01-10T23:53:39Z"       diff
cluster-openshift-controller-manager-operator            091d03d6    "2023-01-12T23:50:12Z"            7fde6865    "2022-12-02T15:15:20Z"       diff
cluster-kube-scheduler-operator                          1ccee33e    "2023-01-16T15:16:15Z"            64f1a6fe    "2023-01-09T19:52:43Z"       diff
machine-config-operator                                  a25109bd    "2023-01-14T00:32:22Z"            1c812547    "2023-01-11T06:01:22Z"       diff
cluster-dns-operator                                     25120536    "2023-01-13T08:37:19Z"            c136fe38    "2022-10-21T03:28:52Z"       diff
cluster-ingress-operator                                 81ee72a0    "2023-01-13T22:17:01Z"            60b2d322    "2022-12-19T14:17:55Z"       diff
service-ca-operator                                      b89fdce3    "2022-11-04T17:20:00Z"            b89fdce3    "2022-11-04T17:20:00Z"       
cluster-network-operator                                 dd2dab27    "2023-01-16T15:04:34Z"            30daae94    "2023-01-09T14:43:38Z"       diff
```

I'm wondering if we should add some logic to keep them synced? Maybe matching closest arm nightly to latest amd nightly.

Reply:
I've updated the script to diff the repos for embedded components separately from each of the image architectures. e7ecf660d133faeb7dcb1368b42a81412eb04112 only shows output for 1 changed image, so it's not that interesting.

Let's move the conversation about whether we need to be smarter about picking images as rebase targets to slack or one of the team meetings.